table/block_based/block.cc

```diff
@@ -128,17 +128,14 @@ struct DecodeKeyV4 {
 };
 
 void DataBlockIter::Next() {
-  assert(Valid());
```

Comment:
This is removed because `Next()` is now used to initialize block iterators ( we call it from `BlockIter::ScanAfterBinarySeek()` where we don't know whether the underlying call will be `ParseNextDataKey()` or `ParseNextIndexKey()`).